src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private static final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int numberRandom = random.nextInt(5);
+        String color = String.valueOf(colorSupplier.getRandomColor());
+
+        switch (numberRandom) {
+            case 0:
+                double side = random.nextDouble() * 10 + 1;
+                return new Square(color, side);
```

Comment:
It is better to move to the private method
```suggestion
                return prepareSquare(color);
```